tests/test_charm.py

```diff
@@ -156,8 +156,6 @@ def test_action_kubeconfig(self, _open, _yaml_dump, _check_call, _check_output):
         )
         self.assertEqual(_check_call.call_args_list, [call(["chown", "-R", "ubuntu:ubuntu", "/home/ubuntu/config"])])
 
-        _yaml_dump.assert_called_once_with(
-            {"clusters": [{"cluster": {"server": "https://public-address:16443"}}]}, _open()
-        )
+        _yaml_dump.assert_called_once_with({"clusters": [{"cluster": {"server": "https://192.0.2.1:16443"}}]}, _open())
```

Comment:
I'd like to see a similar test with an IPv6 address (and perhaps a hostname? though i doubt Juju returns a hostname for a `public-address` in any of the supported clouds.

Reply:
I've added a test for the IPv6 case.